src/app/main-page/main-page.controller.js

```diff
@@ -0,0 +1,33 @@
+export default class MainPageController {
+    constructor($state, lastSearchesService, nearbyCitiesService) {
+        this.$state = $state;
+        this.lastSearchesService = lastSearchesService;
+        this.nearbyCitiesService = nearbyCitiesService;
+        this.nearbyCities = [];
+    }
+
+    $onInit() {
+        this.lastSearches = this.lastSearchesService.getCities();
+        if (this.$state.params.city) {
+            this.param = this.$state.params.city;
+        }
+        this.lastSearchesService.saveCity(this.param, this.lastSearchesService.getCities());
+        this.$state.go('mainPage', {
```

Comment:
why this is necessary?? I mean, the user is already in the mainPage with the city param.